.github/workflows/release-new-version.yml

```diff
@@ -0,0 +1,119 @@
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Run ESLint
+        run: npm run lint
+
+      - name: Run Prettier
+        run: npm run prettier
+
+  build_and_push:
+    runs-on: ubuntu-latest
+    needs: format_and_lint
+    strategy:
+      matrix:
+        build-type: [production, staging]
+        include:
+          - build-type: production
+            tag_suffix: prod
+            environment: production-fidl
+
+          - build-type: staging
+            tag_suffix: stg
+            environment: staging-fidl
+
+    continue-on-error: false
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Cache Docker
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-docker-build-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-docker-build-
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-access-key-id: ${{ secrets[matrix.environment].AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets[matrix.environment].AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Login to Amazon ECR
+        uses: aws-actions/amazon-ecr-login@v1
+        with:
+          mask-password: 'true'
+          registry-type: public
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          flavor: |
+            latest=false
+            suffix=-${{ matrix.tag_suffix }}
+          images: public.ecr.aws/f4h6r4m9/${{ env.ECR_REPOSITORY }}
+          tags: |
+            type=semver,pattern={{version}},value=v${{ inputs.version }},enable=${{inputs.version != ''}}
+            type=raw,value={{branch}},enable=${{inputs.version != ''}}
+            type=ref,event=pr,pattern={{branch}}
+
+      - name: Build tag
+        uses: docker/build-push-action@v6
+        env:
+          NEXT_PUBLIC_API_URL: ${{ vars[matrix.environment].NEXT_PUBLIC_API_URL }}
+          NEXT_PUBLIC_DMOB_API_KEY: ${{ vars[matrix.environment].NEXT_PUBLIC_DMOB_API_KEY }}
+          NEXT_PUBLIC_MODE: ${{ vars[matrix.environment].NEXT_PUBLIC_MODE }}
+          NEXT_PUBLIC_NODE_ADDRESS: ${{ vars[matrix.environment].NEXT_PUBLIC_NODE_ADDRESS }}
+          NEXT_PUBLIC_NODE_TOKEN: ${{ vars[matrix.environment].NEXT_PUBLIC_NODE_TOKEN }}
+          NEXT_PUBLIC_DMOB_API_URL: ${{ vars[matrix.environment].NEXT_PUBLIC_DMOB_API_URL }}
+          NEXT_PUBLIC_GLIF_URL: ${{ vars[matrix.environment].NEXT_PUBLIC_GLIF_URL }}
+          NEXTAUTH_URL: ${{ vars[matrix.environment].NEXTAUTH_URL }}
+          NEXTAUTH_SECRET: ${{ secrets[matrix.environment].NEXTAUTH_SECRET }}
+          GITHUB_ID: ${{ secrets[matrix.environment].GH_ID }}
+          GITHUB_SECRET: ${{ secrets[matrix.environment].GH_SECRET }}
```

Comment:
Don't use real secrets for non-`NEXT_PUBLIC` secrets during build. They shouldn't end up in public image by accident.
I use these placeholders instead for build:
```
NEXTAUTH_SECRET=placeholder
GITHUB_ID=placeholder
GITHUB_SECRET=placeholder
```

`NEXTAUTH_URL` can stay with real var, as it's not sensitive.